whisper/whisperv6/peer.go

```diff
@@ -226,10 +223,18 @@ func (peer *Peer) notifyAboutBloomFilterChange(bloom []byte) error {
 }
 
 func (peer *Peer) bloomMatch(env *Envelope) bool {
-	if peer.bloomFilter == nil {
-		// no filter - full node, accepts all envelops
-		return true
-	}
+	return peer.fullNode || bloomFilterMatch(peer.bloomFilter, env.Bloom())
+}
 
-	return bloomFilterMatch(peer.bloomFilter, env.Bloom())
+func (peer *Peer) setBloomFilter(bloom []byte) {
+	peer.bloomFilter = bloom
+	peer.fullNode = isFullNode(bloom)
+	if peer.fullNode {
+		if peer.bloomFilter == nil {
```

Comment:
I see a problem here, because I see that you are initializing it to all 1s when you set it. That means that, at some point, there could be someone trying to read the bloom filter before someone has initialized it. So I believe there should be a `getBloomFilter` as well.